parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

```diff
@@ -765,6 +801,36 @@ public static Statistics toParquetStatistics(
     return formatStats;
   }
 
+  private static BoundingBox toParquetBoundingBox(org.apache.parquet.column.statistics.geometry.BoundingBox bbox) {
+    // Check if any of the required bounding box is valid.
+    if (!bbox.isValid() || bbox.isEmpty()) {
+      // According to the Thrift-generated class, these fields are marked as required and must be set explicitly.
+      // If any of them is NaN, it indicates the bounding box is invalid or uninitialized,
+      // so we return null to avoid creating a malformed BoundingBox object that would later fail serialization
+      // or validation.
+      return null;
+    }
+
+    // Now we can safely create the BoundingBox object
+    BoundingBox formatBbox = new BoundingBox();
+    formatBbox.setXmin(bbox.getXMin());
+    formatBbox.setXmax(bbox.getXMax());
+    formatBbox.setYmin(bbox.getYMin());
+    formatBbox.setYmax(bbox.getYMax());
+
+    if (Double.isInfinite(bbox.getZMin()) && Double.isInfinite(bbox.getZMax())) {
+      formatBbox.setZmin(bbox.getZMin());
+      formatBbox.setZmax(bbox.getZMax());
+    }
+
+    if (Double.isInfinite(bbox.getMMin()) && Double.isInfinite(bbox.getMMax())) {
+      formatBbox.setMmin(bbox.getMMin());
+      formatBbox.setMmax(bbox.getMMax());
+    }
```

Comment:
```suggestion
    if (!Double.isInfinite(bbox.getZMin()) && !Double.isInfinite(bbox.getZMax()) && !Double.isNaN(bbox.getZMin()) && !Double.isNaN(bbox.getZMax())) {
      formatBbox.setZmin(bbox.getZMin());
      formatBbox.setZmax(bbox.getZMax());
    }

    if (!Double.isInfinite(bbox.getMMin()) && !Double.isInfinite(bbox.getMMax()) && !Double.isNaN(bbox.getMMin()) && !Double.isNaN(bbox.getMMax())) {
      formatBbox.setMmin(bbox.getMMin());
      formatBbox.setMmax(bbox.getMMax());
    }
```

Shouldn't we write any inf or nan to thrift?